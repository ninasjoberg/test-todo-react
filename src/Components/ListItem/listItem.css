
li {
    list-style: none;
}

.item {
    display: flex;
}

.deleteButton {
    margin-left: 10px;
}