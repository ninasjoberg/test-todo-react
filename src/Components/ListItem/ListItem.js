import React from 'react';
import PropTypes from 'prop-types';
import './listItem.css';


export default function ListItem(props) {
    const {
        item, onDelete, onDone, type,
    } = props;

    return (
        <div>
            <li className="item">
                <p>{item.text}</p>
                <div className="buttons">
                    {(type === 'todo') &&
                        <button className="complete-button" onClick={() => { onDone(item); }}>Done</button>
                    }
                    <button className="delete-button" onClick={() => { onDelete(item); }}>Delete</button>
                </div>
            </li>
        </div>
    );
}


ListItem.propTypes = {
    item: PropTypes.shape({
        text: PropTypes.string,
        id: PropTypes.number,
    }).isRequired,
    type: PropTypes.string.isRequired,
};

