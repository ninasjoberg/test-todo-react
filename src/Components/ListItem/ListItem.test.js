import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ListItem from './ListItem';


const item = {
    text: 'koka kaffe',
    id: 4,
};

let fakefunc;

beforeEach(() => {
    fakefunc = jest.fn();
});


it('renders an li-tag when tehre is a todo', () => {
    const wrapper = shallow(<ListItem type="todo" item={item} />);
    expect(wrapper.find('li')).toBeDefined();
});


it('renders a complete-button if typ=todo', () => {
    const type = 'todo';
    const wrapper = shallow(<ListItem type={type} item={item} />);
    expect(wrapper.find('.complete-button').exists()).toBeTruthy();
});


it('does not render a complete-button if typ=done', () => {
    const type = 'done';
    const wrapper = shallow(<ListItem type={type} item={item} />);
    expect(wrapper.find('.complete-button').exists()).toBeFalsy();
});


it('simulate onclick success delete-button', () => {
    const type = 'toDo';
    const wrapper = mount(<ListItem type={type} item={item} onDelete={fakefunc} />);
    wrapper.find('.delete-button').simulate('click');
    expect(fakefunc).toHaveBeenCalled();
});


it('simulate onclick success doneButton', () => {
    const type = 'todo';
    const wrapper = mount(<ListItem type={type} item={item} onDone={fakefunc} />);
    wrapper.find('.complete-button').simulate('click');
    expect(fakefunc).toHaveBeenCalled();
});


// Snapshot test:

it('should match snapshot', () => {
    const type = 'todo';
    const tree = renderer
        .create(<ListItem type={type} item={item} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

