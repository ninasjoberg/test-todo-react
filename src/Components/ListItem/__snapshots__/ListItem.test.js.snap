// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <li
    className="item"
  >
    <p>
      koka kaffe
    </p>
    <div
      className="buttons"
    >
      <button
        className="complete-button"
        onClick={[Function]}
      >
        Done
      </button>
      <button
        className="delete-button"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </li>
</div>
`;
